Source/JavaScriptCore/runtime/JSONObject.cpp

```diff
@@ -627,6 +637,517 @@ bool Stringifier::Holder::appendNextProperty(Stringifier& stringifier, StringBui
+                if (remainingCapacity() < needComma + 1 + nameLength + 2) {
+                    recordBufferFull();
+                    return false;
+                }
+                if (needComma)
+                    m_buffer[m_length++] = ',';
+                m_buffer[m_length] = '"';
+                auto* characters = name.characters8();
+                for (unsigned i = 0; i < nameLength; ++i) {
+                    auto character = characters[i];
+                    if (WTF::escapedFormsForJSON[character]) {
+                        recordFailure("property name character needs escaping"_s);
+                        return false;
+                    }
+                    m_buffer[m_length + 1 + i] = character;
+                }
+                m_buffer[m_length + 1 + nameLength] = '"';
+                m_buffer[m_length + 1 + nameLength + 1] = ':';
+                m_length += 1 + nameLength + 2;
+                append(object.getDirect(entry.offset()));
+                return !haveFailure();
+            });
+            if (haveFailure())
+                return;
+        } else {
+#if !FAST_STRINGIFY_GETTER_EXPERIMENT
+            recordFastPropertyEnumerationFailure(object);
+            return;
+#else
+            auto scope = DECLARE_THROW_SCOPE(m_vm);
+            PropertyNameArray names(m_vm, PropertyNameMode::Strings, PrivateSymbolMode::Exclude);
+            object.methodTable()->getOwnPropertyNames(&object, &m_globalObject, names, DontEnumPropertiesMode::Exclude);
+            if (UNLIKELY(scope.exception())) {
+                scope.clearException();
+                recordFailure("getOwnPropertyNames exception"_s);
+                return;
+            }
+            for (auto& name : names) {
+                if (!name.string().is8Bit()) {
+                    recordFailure("16-bit property name"_s);
+                    return;
+                }
+                bool needComma = m_buffer[m_length - 1] != '{';
+                unsigned nameLength = name.length();
+                if (remainingCapacity() < needComma + 1 + nameLength + 2) {
+                    recordBufferFull();
+                    return;
+                }
+                if (needComma)
+                    m_buffer[m_length++] = ',';
+                m_buffer[m_length] = '"';
+                auto* characters = name.string().characters8();
+                for (unsigned i = 0; i < nameLength; ++i) {
+                    auto character = characters[i];
+                    if (mayNeedJSONEscaping(character)) {
+                        recordFailure("property name character needs escaping"_s);
+                        return;
+                    }
+                    m_buffer[m_length + 1 + i] = character;
+                }
+                m_buffer[m_length + 1 + nameLength] = '"';
+                m_buffer[m_length + 1 + nameLength + 1] = ':';
+                m_length += 1 + nameLength + 2;
+                scope.assertNoException();
```

Comment:
Let's remove this. Unfortunately, this is some kind of stale function since exception can happen at any time if there is scope (termination exception can be thrown from the watchdog thread asynchronously.)

Reply:
If you don't want interference from the `TerminationException`, declare a `DeferTerminationForAWhile` scope at top of your scope e.g. in `fastStringify`.  This is ok to do if your function is bounded i.e. it won't be calling arbitrary code (like calling back into JS) and is guaranteed to terminate in some smallish finite time in the near future, thereby letting someone else service the termination request later instead of you.

That said, you're in `FastStringifier`, and I would expect side-effect free code to never throw, and therefore, won't need exception checks.  No exception checks == no chance for a termination request to be serviced, hence, should not see a `TerminationException`.

Reply:
Obviously, the patch is doing exception checks (though I haven't grok why this is needed yet).  So TerminationExceptions are possible.